pkg/network/controllers/vm_test.go

```diff
@@ -244,6 +246,45 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(iface.State).NotTo(Equal(v1.InterfaceStateAbsent))
 	})
 
+	DescribeTable("sync updates link state correctly", func(state v1.InterfaceState) {
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(
+			clientset,
+			stubPodGetter{pod: nil},
+		)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+		defaultIfaceName := vmi.Spec.Domain.Devices.Interfaces[0].Name
+		vmi = setLinkState4NetInterface(vmi, defaultIfaceName, state)
```

Comment:
In case the interface has to be at some state at the begging of the test:
```go
iface := libvmi.InterfaceDeviceWithMasqueradeBinding()
iface.State = state

vmi := libvmi.New(
			libvmi.WithInterface(iface),
                        ...
            )
```

Its clearer and does less work.